limitQueryAmountMiddleware.go

```diff
@@ -0,0 +1,75 @@
+package graphql
+
+import (
+	"context"
+
+	gql "github.com/99designs/gqlgen/graphql"
+	"github.com/vektah/gqlparser/v2/ast"
+)
+
+const (
+	sameOperationsDefaultThreshold = 2
+	allOperationsDefaultThreshold  = 10
+)
+
```

Comment:
We should ensure that we implement the func/interface from the gql package correctly:

`var _ gql.OperationMiddleware = LimitQueryAmountMiddleware(nil)`